app/views/collapse_expand/_collapse_expand.js.erb

```diff
@@ -0,0 +1,119 @@
+<script type="text/javascript">
+let collapsedIds = '<%= @issue.collapsed_ids.to_s %>';
+let issueId = '<%= @issue.id %>';
+
+function collapseExpand(element) {
+  // Helper function to handle the display logic
+  function toggleIssues(targetIssue, targetIssueIdntLevel, show) {

```

Comment:
You can simplify this code by using `JQuery` which is alreay included by redmine, so you just need to call the code.

In my opinion, I think you could use jquery to get all the issues that are after the index you received, and have an indent bigger the the element by using `filter`, as in https://stackoverflow.com/a/2613671

Just get the div for all the issues, then all the tr that have than info, and alter showing.

Should greatly simplify the code.